manifests/init.pp

```diff
@@ -375,6 +383,52 @@
     ensure => $gp_package_ensure,
   }
 
+  $once_file = '/etc/opt/quest/vas/puppet_joined'
+
+  # no run if undef!
+  # We should probably have better sanity checks for $realm parameter instead of this.
+  if $realm != undef {
+    # So we use the fact vas_domain to identify if vas is already joined to a AD
+    # server. It will make sure and check that ::vas_domain is not undef before doing this
+    # to prevent something from happening at first run.
+    # If the vas_domain fact is not the same as the realm specified in hiera it
+    # will then check if the domain_change_real parameter is set to true. If it is
+    # it will join the domain with help of the lastjoin file.
+    # If the domain_change_real fact is false, it will fail the compilation and warn
+    # of the mismatching realm.
+    if $::vas_domain != $realm and $::vas_domain != undef  {
+      if $domain_change_real == true {
+        exec { 'vas_change_domain':
+          # This command executes the result of the sed command, puts the log from
+          # the unjoin command into a log file and removes the once file to allow
+          # the  vas_inst command to join the new AD server.
+          # This is how the join command is built up by the vas module.
+          # ${vastool_binary} -u ${username} -k ${keytab_path} -d3 join -f ${workstation_flag} -c ${computers_ou} ${user_search_path_parm} ${group_search_path_parm} ${upm_search_path_parm} -n ${vas_fqdn}
+          # The sed regex will save everything up to but not including the join part.
+          # (${vastool_binary} -u ${username} -k ${keytab_path} -d3 )
+          # It will save the part above and add to the end of it unjoin.
+          # The result would be ${vastool_binary} -u ${username} -k ${keytab_path} -d3 unjoin
+          #
+          # This sed command is required because we need to use the old credentials
+          # and old username to unjoin the currently joined AD.
+          # It could be that you need to use a newly created keytab file for perhaps
+          # the same/new user required to join the new AD Server. So to join the
+          # new AD server we would need updated hiera information for that. Preventing
+          # us from using the new hiera data to unjoin the current AD Server.
+          command  => "$(sed 's/\\(.*\\)join.*/\\1unjoin/' /etc/opt/quest/vas/lastjoin) > /tmp/vas_unjoin.txt 2>&1 && rm -f ${once_file}",
```

Comment:
escaped the escape symbol \  and it works both for the syntax check and the resulting command is the same.